replication.drush.inc

```diff
@@ -0,0 +1,261 @@
+      ),
+      'output-data-type' => 'format-list',
+    ),
+  );
+  $items['stop-replication'] = array(
+    'description' => 'Stop a replication.',
+    'arguments' => array(
+      'source' => dt('Source database.'),
+      'target' => dt('Target database.'),
+    ),
+    'options' => array(
+      'replicator' => array(
+        'description' => dt('The used replicator.'),
+      ),
+    ),
+    'outputformat' => array(
+      'default' => 'key-value',
+      'pipe-format' => 'json',
+      'field-labels' => array(
+        'ok' => 'Status',
+        'no_changes' => 'No changes',
+        'session_id' => 'Session ID',
+        'source_last_seq' => 'Last sequence number',
+        'replication_id_version' => 'Replication protocol version',
+      ),
+      'output-data-type' => 'format-list',
+    ),
+  );
+  $items['active-replication'] = array(
+    'description' => 'Prints information about the specific active replication between target and source databases.',
+    'arguments' => array(
+      'source' => dt('Source database.'),
+      'target' => dt('Target database.'),
+    ),
+    'options' => array(
+      'replicator' => array(
+        'description' => dt('The used replicator.'),
+      ),
+    ),
+    'outputformat' => array(
+      'default' => 'key-value-list',
+      'pipe-format' => 'json',
+      'field-labels' => array(
+        'source' => 'Source',
+        'target' => 'Target',
+        'started_on' => 'Started on',
+        'progress' => 'Progress',
+        'docs_read' => 'Documents read',
+        'docs_written' => 'Documents written',
+        'revisions_checked' => 'Revisions checked',
+        'doc_write_failures' => 'Write failures',
+        'pid' => 'Process ID',
+      ),
+    ),
+  );
+
+  return $items;
+}
+
+/**
+ * Implements drush_hook_COMMAND_validate().
+ */
+function drush_replication_start_replication_validate() {
+  // Array of "Callback arguments" and "command line args".
+  $params = func_get_args();
+  _replication_command_validate($params);
+}
+
+/**
+ * Implements drush_hook_COMMAND_validate().
+ */
+function drush_replication_stop_replication_validate() {
+  // Array of "Callback arguments" and "command line args".
+  $params = func_get_args();
+  _replication_command_validate($params);
+}
+
+/**
+ * Implements drush_hook_COMMAND_validate().
+ */
+function drush_replication_active_replication_validate() {
+  // Array of "Callback arguments" and "command line args".
+  $params = func_get_args();
+  if (isset($params[0]) || isset($params[1])) {
+    _replication_command_validate($params);
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ */
+function drush_replication_start_replication($source, $target) {
+  try {
+    $options = array('dbname' => '');
+    if ($replicator = drush_get_option('replicator')) {
+      $url_parts = _get_url_parts($replicator, TRUE);
+      $options = array_merge($options, $url_parts);
+    }
+    $client = CouchDBClient::create($options);
```

Comment:
This piece of code...

```php
$options = array('dbname' => '');
if ($replicator = drush_get_option('replicator')) {
  $url_parts = _get_url_parts($replicator, TRUE);
  $options = array_merge($options, $url_parts);
}
$client = CouchDBClient::create($options);
```
...is repeated in every command. It would be better to create a `replication_client_factory()` function that creates a new `$client` object that every function can use.